Backend/router/v1/user.js

```diff
@@ -1,98 +1,85 @@
 const express = require('express');
 const router = express.Router();
 const models = require('../../models');
-const crypto = require('crypto');
-const session = require('express-session');
+const crypto = require('crypto')
 
 const MIN_PASSWORD_LENGTH = 6;
 
-router.use(session({
-  key: 'sid',
-  secret: 'secret',
-  resave: false,
-  saveUninitialized: true,
-  cookie: {
-    maxAge: 1000*60*15
-  }
-}));
+module.exports = (passport) => {
+  router.use((req, res, next) => {
+    next();
+  });
 
-router.use((req, res, next) => {
-  console.log("user router");
-  next();
-});
+  router.all('/', (req, res) => {
+    res.status(404);
+  });
 
-router.all('/', (req, res) => {
-  res.send("this is user root");
-});
+  router.get('/signin', (req, res, next) => {
+    passport.authenticate('local', {
+      successRedirect : '/api/v1/user/signin-success',
+      failureRedirect : '/api/v1/user/signin-failure'
+    }) (req,res,next);
+  });
 
-function password_length_check(length){
-  return !(length < MIN_PASSWORD_LENGTH);
-}
+  router.get('/signin-success', (req, res) => {
+    res.status(200);
+    res.json({
+      msg: "signin success"
+    });
+  });
 
-router.post('/signup', (req, res) => {
-  let hashedPassword = crypto.createHash("sha512").update(req.body.password).digest("hex");
+  router.get('/signin-failure', (req, res) => {
+    res.status(401);
+    res.json({
+      msg: "signin failed"
+    });
+  });
 
-  models.user.findAll({
-    where: {
-      email: req.body.email
-    }
-  }).then(result => {
-    if (result == "") {
-      if (password_length_check(req.body.password.length) == false ) {
-        res.status(400).send("Password must be " + MIN_PASSWORD_LENGTH + " over");
-      } else {
-        models.user.create({
-          email: req.body.email,
-          password: hashedPassword,
-        }).then(result => {
-            res.send(
-              "Account successfully created"
-            );
-        });
-      }
+  router.get('/signout', (req, res) => {
+    if(req.isAuthenticated()){
+      res.status(200);
+      req.session.destroy();
+      res.clearCookie('sid');
+      res.json({
+        msg: "signout success"
+      });
+=======
```

Comment:
=====

Reply:
Looks like this is the leftover merge conflict resolution.
Have you guys ever run the app after you merge?